ksql-functional-tests/src/test/java/io/confluent/ksql/test/EndToEndEngineTestUtil.java

```diff
@@ -144,6 +148,24 @@ private static PersistentQueryMetadata buildQuery(
         Optional.of(serviceContext.getSchemaRegistryClient())
     );
 
+    final MetaStore metaStore = ksqlEngine.getMetaStore();
+    for (QueryMetadata queryMetadata: queries) {
+      final PersistentQueryMetadata persistentQueryMetadata
+          = (PersistentQueryMetadata) queryMetadata;
+      final String sinkKafkaTopicName = metaStore.getSource(
+          persistentQueryMetadata.getSinkNames().iterator().next()).getKafkaTopicName();
```

Comment:
we should change this to `Iterables.getOnlyElement(metadata.getSinkNames())` so that if we every change it to have multiple sinks this will fail violently